RELEASES.md

```diff
@@ -197,11 +199,107 @@ To create a minor release:
+code is stable and reliable. Broken tests are fixed, flakey-tests are remedied,
+end-to-end test failures are thoroughly diagnosed and all efforts of the team
+are aimed at improving the quality of the code. During this period, the upgrade
+harness tests are run repeatedly and a variety of in-house testnets are run to
+ensure Tendermint functions at the scale it will be used by application
+developers and node operators.
+
+### Nightly End-To-End Tests
+
+The Tendermint team maintains [a set of end-to-end
+tests](https://github.com/tendermint/tendermint/blob/master/test/e2e/README.md#L1)
+that run each night on the latest commit of the project and on the code in the
+tip of each backport branch. These tests start a network of containerized
+Tendermint processes and run automated checks that the network functions as
+expected in both stable and unstable conditions. During the feature freeze,
+these tests are run nightly and must pass consistently for a release of
+Tendermint to be considered stable.
+
+### Upgrade Harness
+
+The Tendermint team is creating an upgrade test harness to exercise the
+workflow of stopping an instance of Tendermint running one version of the
+software and starting up the same application running the next version. To
+support upgrade testing, we will add the ability to terminate the Tendermint
+process at specific pre-defined points in its execution so that we can verify
+upgrades work in a representative sample of stop conditions.
+
+### Large Scale Testnets
+
+The Tendermint end-to-end tests run a small network (~10s of nodes) to exercise
+basic consensus interactions. Real world deployments of Tendermint often have over 
+a hundred nodes just in the validator set, with many others acting as full
+nodes and sentry nodes. To gain more assurance before a release, we will also run
+larger-scale test networks to shake out emergent behaviors at scale.
+
+Large-scale test networks are run on a set of virtual machines (VMs). Each VM
+is equipped with 4 Gigabytes of RAM and 2 CPU cores. The network runs a very
+simple key-value store application. The application adds artificial delays to
+different ABCI calls to simulate a slow application. During each test net, the
+following metrics are monitored and collected on each node:
+
+* Tendermint rounds per height
+* Peers connected
+* Memory resident set size
+* CPU utilization
+* Blocks produced per minute
+* Seconds for each step of consensus (Propose, Prevote, Precommit, Commit)
+* Latency to receive each block proposal
+
+For these tests we intentionally target low-powered host machines (with low core
+counts and limited memory) to ensure we observe similar kinds of resource contention 
+and limitation that real-world  deployments of Tendermint experience in production. 
+
+#### 200 Node Testnet
+
+To test the stability and performance of Tendermint in a real world scenario,
+a 200 node test network is run. The network comprises 5 seed nodes, 100
```

Comment:
I think I agree with Evan. I know bigger testnets will have higher cost in terms of $$$ and time but we should test our product _at least_ in the same scalability conditions as it will be used on the field. I know it's not guarantee of catching the same problems (especially because the App will be different) but it's worth testing those conditions.

Reply:
Added language to indicate this should be considered when performance critical systems are changed.

Reply:
I don't want to arrive late and reopen conversations, but following the same line of my previous comments, I believe that the level of detail on this document should be higher. In this case, in particular, we can mention that we run a testnet with a number of nodes (validators, non-validators, etc.) which is in line with Cosmos Hub, for instance. If the number of nodes in that network increases between releases, this document remain updated. 

In my opinion, the precise number of nodes using the testnet is a detail that we provide at each release, not in document that governs every release.